pandas-stubs/core/frame.pyi

```diff
@@ -223,9 +220,11 @@ class DataFrame:
     def query(self, expr: _str) -> DataFrame: ...
     def reindex(self, index: Index) -> DataFrame: ...
     @overload
-    def rename(self, mapper: Callable, axis: _AxisType = ...) -> DataFrame: ...
+    def rename(self, mapper: Union[Dict, Callable], axis: _AxisType = ...) -> DataFrame: ...
     @overload
-    def rename(self, columns: Dict[_str, _str]) -> DataFrame: ...
+    def rename(self, columns: Union[Dict, Callable]) -> DataFrame: ...
+    @overload
+    def rename(self, index: Union[Dict, Callable]) -> DataFrame: ...
```

Comment:
Can we [force named arguments](https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html) here?

```suggestion
    def rename(self, *, index: Union[Dict, Callable]) -> DataFrame: ...
```

Otherwise this is a bit confusing.

Reply:
You're right, added this for both index and columns argument